src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsHostBuilderExtensions.cs

```diff
@@ -135,6 +140,38 @@ public static IHostBuilder ConfigureWebJobs(this IHostBuilder builder, Action<Ho
             return builder;
         }
 
+        /// <summary>
+        /// Configures the specified <see cref="IHostBuilder"/> as a scale manager host.
+        /// </summary>
+        /// <param name="builder">The <see cref="IHostBuilder"/> to configure.</param>
+        /// <param name="configure">Configuration action to perform as part of service configuration.</param>
+        /// <param name="configureScaleOptions">Configuration action for <see cref="ScaleOptions"/>.</param>
+        /// <returns>The <see cref="IHostBuilder"/>.</returns>
+        public static IHostBuilder ConfigureWebJobsScale(this IHostBuilder builder,
+            Action<HostBuilderContext, IWebJobsBuilder> configure,
+            Action<ScaleOptions> configureScaleOptions)
+        {
+            builder.ConfigureServices((context, services) =>
+            {
+                WebJobsBuilder webJobsBuilder = new WebJobsBuilder(services);
+                configure?.Invoke(context, webJobsBuilder);
+
+                if (configureScaleOptions != null)
+                {
+                    services.Configure(configureScaleOptions);
+                }
+
+                services.AddFeatureManagement();
+                services.TryAddSingleton<IScaleMonitorManager, ScaleMonitorManager>();
+                services.TryAddSingleton<ITargetScalerManager, TargetScalerManager>();
+                services.TryAddSingleton<IScaleManager, ScaleManager>();
+                services.TryAddSingleton<IHostedService, ScaleMonitorService>();
+                services.TryAddSingleton<IPrimaryHostStateProvider>(new PrimaryHostStateProvider() { IsPrimary = true });
```

Comment:
Not clear to me you should be hardcoding IsPrimary like this, as opposed to also registering PrimaryHostCoordinator as is done normally. Then IsPrimary is based on existing logic and use of IDistributedLockManager. The main point is that we need to think about Functions Host usages and host builder ordering to ensure we're using the right services.

Reply:
Discussed offline - `PrimaryHostStateProvider` is a minimal interface needed for `ScaleMonitorService`

Reply:
How does this layer with Azure Functions usage though? In Functions we're calling ConfigureWebJobsScale (I see in your PR) so it'll register this dummy impl. when we want an actual impl.

Reply:
In functions I checked that by the time of executing `ConfigureWebJobsScale` `PrimaryHostStateProvider` has been registered so `TryAddSingleton` will not add the dummy impl to the host.